### PR TITLE
table & model changes for group managers with permission to edit members

### PR DESCRIPTION
**Pull request by jmay**
Extracted the just the back-end parts from https://github.com/discourse/discourse/pull/3004

Includes API changes on `/groups/xxx` (not `/admin/groups/xxx`)

### CODE REVIEW

**Comment by jmay** [comment]
Thanks for the feedback, all applied.

**Comment by jmay** [comment]
Travis shows red, but the failure seems to be a network issue when running the javascript suite, not anything with the app. Is there a way to trigger a test rerun?

**Comment by jmay** [comment]
Maybe just merging master will give it a kick?

File: app/models/group.rb [hunk 1]
```diff
@@ -277,6 +280,10 @@ def remove(user)
     self.group_users.where(user: user).each(&:destroy)
   end
 
+  def appoint_manager(user)
```
Comment (ZogStriP) [inline app/models/group.rb]: Not sure that method is needed since it's only used in tests.

**Comment by ZogStriP** [comment]
Just added a comment. Once fixed, don't forget to squash the commits into 1 and I'll get that merged ;)

File: app/models/group.rb (hunk 1 above)
Comment (jmay) [inline app/models/group.rb]: I put that in so there is a place for the controller to invoke once there is a UI for this (I'm working on it, slowly). Not sure where the best separator is between back-end and front-end stuff, and I didn't want to touch the front-end with this patch.

Comment (ZogStriP) [inline app/models/group.rb]: That's fine. Can you squash the commits into 1 and then I'll merge that ;)

**Comment by jmay** [comment]
There we go. *rebase is the mind-killer. rebase is the little-death that brings total obliteration.* I will face my fear.

**Pull request merged**